applications/GeoMechanicsApplication/custom_utilities/stress_strain_utilities.cpp

```diff
@@ -57,6 +58,55 @@ double StressStrainUtilities::CalculateMeanStress(const Vector& StressVector)
     return CalculateTrace(StressVector) / (StressVector.size() == 3 ? 2.0 : 3.0);
 }
 
+double StressStrainUtilities::CalculateLodeAngle(const Vector& StressVector)
+{
+    KRATOS_ERROR_IF(StressVector.size() < 4);
+
+    const double p                   = CalculateMeanStress(StressVector);
+    const double q                   = CalculateVonMisesStress(StressVector);
+    Matrix       local_stress_tensor = MathUtils<double>::StressVectorToTensor(StressVector);
+    Matrix       sigma_princi;
+    Matrix       eigen_vectors;
+    MathUtils<double>::GaussSeidelEigenSystem(local_stress_tensor, eigen_vectors, sigma_princi, 1.0e-16, 20);
+    const double numerator = (sigma_princi(0, 0) - p) * (sigma_princi(1, 1) - p) * (sigma_princi(2, 2) - p);
+    if (std::abs(numerator) < 1.0E-12) return 0.;
+    return std::asin((-27. / 2.) * numerator / (q * q * q)) / 3.0;
+}
+
+double StressStrainUtilities::CalculateMCShearCapacity(const Vector& StressVector, double C, double Phi)
+{
+    KRATOS_ERROR_IF(StressVector.size() < 4);
+
+    const double q_mc = CalculateQMC(StressVector, C, Phi);
+    const double q    = CalculateVonMisesStress(StressVector);
+
+    return q / q_mc;
+}
+
+double StressStrainUtilities::CalculateQMC(const Vector& StressVector, double C, double Phi)
+{
+    const double denominator = CalculateDenominator(StressVector, Phi);
+    const double p           = -CalculateMeanStress(StressVector);
+    return 3. * (p * sin(Phi) + C * cos(Phi)) / denominator;
```

Comment:
I would suggest to use the mathematical functions defined in the `std` namespace. For instance, see [this StackOverflow question](https://stackoverflow.com/questions/76282274/should-i-use-std-when-calling-cmath-functions) for more information.
```suggestion
    return 3. * (p * std::sin(Phi) + C * std::cos(Phi)) / denominator;
```

Reply:
Done on all sin and cos